### PR TITLE
Block exporter view

### PR DESCRIPTION
**Pull request by svouse**
Basic framework for UI support of block exporting. Marks the official end of block_exporter refactoring.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-devtools/79)

### CODE REVIEW

**Comment by chynu** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/view/app_view.js, line 218 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/79#-KoO5AbdJYY8TFLoDWhp:-KoO5AbdJYY8TFLoDWhq:b-z928jy) ([raw file](https://github.com/google/blockly-devtools/blob/f6ab832c8cb4a4f9a5b3ae5300676ee9de7a7082/src/view/app_view.js#L218)):*
> ```JavaScript
>    * Calls response to selecting export project in menu.
>    */
>   exportProject(){
> ```

Nit but space between close parentheses and open bracket (?) 0:

---

*[src/view/navigation_tree.js, line 185 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/79#-KoO7IALMpz3WMSrt7Is:-KoO7IALMpz3WMSrt7It:b-iv1lam) ([raw file](https://github.com/google/blockly-devtools/blob/f6ab832c8cb4a4f9a5b3ae5300676ee9de7a7082/src/view/navigation_tree.js#L185)):*
> ```JavaScript
>         action: //TODO: add function for exporting associated element
>       }
>     }
> ```

Add semicolon?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/79)*

File: src/view/app_view.js [hunk 1]
```diff
@@ -202,6 +213,41 @@ class AppView {
   }
 
   /**
+   * Calls response to selecting export project in menu.
+   */
+  exportProject(){
+    // TODO: Fill in action.
```
Comment (AnmAtAnm) [inline src/view/app_view.js]: Console.log("unimplemented AppView.exportProject()") here and similarly below.

**Comment by svouse** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/view/app_view.js, line 218 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/79#-KoO5AbdJYY8TFLoDWhp:-KoO9QWlMuh5X_5yJkWu:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/f6ab832c8cb4a4f9a5b3ae5300676ee9de7a7082/src/view/app_view.js#L218)):*
<details><summary><i>Previously, chynu (celine choo) wrote…</i></summary><blockquote>

Nit but space between close parentheses and open bracket (?) 0:
</blockquote></details>

Done.

---

*[src/view/navigation_tree.js, line 185 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/79#-KoO7IALMpz3WMSrt7Is:-KoO9awnd0B_b7s_EQbl:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/f6ab832c8cb4a4f9a5b3ae5300676ee9de7a7082/src/view/navigation_tree.js#L185)):*
<details><summary><i>Previously, chynu (celine choo) wrote…</i></summary><blockquote>

Add semicolon?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/79)*

File: src/view/navigation_tree.js [hunk 2]
```diff
@@ -162,4 +162,28 @@ class NavigationTree {
...
+  /**
+   * Creates menu for right click functionality.
+   * @return {!Object} The right click menu for the nodes in the tree.
+   * //TODO: add right click functionality to tree
+   */
+   createMenu() {
+    const items = {
+      renameElement : {
+        label: "Rename",
+        action: //TODO: add function for renaming node and associated element
```
Comment (AnmAtAnm) [inline src/view/navigation_tree.js]: Does this parse without a value for action?  I'd at least put in a null value.  Better yet, a () => Console.log(..).

Same below.

File: src/view/app_view.js (hunk 1 above)
Comment (svouse) [reply src/view/app_view.js]: Done.

File: src/view/navigation_tree.js (hunk 2 above)
Comment (svouse) [reply src/view/navigation_tree.js]: Done.

**Review by AnmAtAnm — APPROVED** [review]
LGTM

**Pull request merged**